src/components/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,131 @@
+import {
+  FC,
+  useState,
+  useEffect,
+  FormEvent,
+  useRef,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo,
+  loadingTodosId: number[],
+  onDelete: (todoId: number) => void,
+  onUpdate: (todoId: number, updatedTodo: Partial<Todo>) => void,
+};
+
+export const TodoInfo: FC<Props> = ({
+  todo,
+  loadingTodosId,
+  onDelete,
+  onUpdate,
+}) => {
+  const { title, completed, id } = todo;
+  const [todoTitle, setTodoTitle] = useState(title);
+  const [isEditing, setIsEditing] = useState(false);
+
+  const handleDelete = async (todoId: number) => {
+    await onDelete(todoId);
+  };
+
+  const handleToggle = async (todoId: number) => {
```

Comment:
Consider using useCallback for handleToggle to prevent unnecessary re-rendering of the component when the function is updated. 🔄